tests/test_instrument_transfer.py

```diff
@@ -0,0 +1,453 @@
+    archive_dirs = [
+        "/data/nosync",
+        "/data/nosync/experiment/sample",
+        "/data/nosync/experiment/sample/run",
+    ]
+    neighbor_dirs = [
+        None,
+        "/data/experiment/sample/run2",
+        "/data/experiment/sample2",
+    ]
+
+    for archive_dir in archive_dirs:
+        for neighbor_dir in neighbor_dirs:
+            # Set up tmp dir
+            tmp = tempfile.TemporaryDirectory()
+            tmp_path = tmp.name
+
+            # Create run dir
+            experiment_path = tmp_path + "/data/experiment"
+            sample_path = experiment_path + "/sample"
+            run_path = sample_path + f"/{DUMMY_RUN_NAME}"
+            os.makedirs(run_path)
+
+            # Create neightbor dir, if any
+            if neighbor_dir:
+                neighbor_path = tmp_path + neighbor_dir
+                os.mkdir(neighbor_path)
+
+            # Create archive dir
+            archive_path = tmp_path + archive_dir
+            os.makedirs(archive_path)
+
+            # Execute code
+            instrument_transfer.archive_finished_run(run_path, archive_path)
+
+            # Assert run is moved to archive dir
+            assert os.path.exists(archive_path + f"/experiment/sample/{DUMMY_RUN_NAME}")
+
+            # Assert run is removed from original location
+            assert not os.path.exists(run_path)
+
+            # Assert experiment and sample dirs are removed if empty
+            if neighbor_dir:
+                assert os.path.exists(neighbor_path)
+                if neighbor_dir == "/data/experiment/sample/run2":
+                    assert os.path.exists(sample_path)
+                else:
+                    assert not os.path.exists(sample_path)
+            else:
+                assert not os.path.exists(experiment_path)
+
+            tmp.cleanup()
+
+
+def create_logs() -> list:
+    tmp = tempfile.TemporaryDirectory()
+
+    # For each instrument position dir and corresponding flowcell name...
+    for position_dir, flowcell_name in {
+        "1A": "PAM12345",
+        "MN19414": "FLG12345",
+    }.items():
+        # --> Create position dir
+        position_path = tmp.name + f"/log/{position_dir}"
+        os.makedirs(position_path)
+
+        # --> Populate each position dir with two log files
+        for file_n in range(1, 3):
+            # --> Populate each log file with two entries
+            lines = []
+            for entry_n in range(1, 3):
+                lines += [
+                    "2023-10-31 15:12:54.1354    INFO: mux_scan_result (user_messages)",
+                    f"    flow_cell_id: {flowcell_name}",
+                    f"    num_pores: {file_n}",
+                    f"    total_pores: {entry_n}",
+                ]
+
+            with open(position_path + f"/control_server_log-{file_n}.txt", "w") as file:
+                file.write("\n".join(lines))
+
+    # Run code
+    logs = instrument_transfer.parse_position_logs(tmp.name + "/log")
```

Comment:
I think I would move this line to the two tests since it's this code that is actually tested right?